Makefile.pre.in

```diff
@@ -2570,6 +2583,14 @@ libinstall:	all $(srcdir)/Modules/xxmodule.c
 	$(INSTALL_DATA) `cat pybuilddir.txt`/_sysconfigdata_$(ABIFLAGS)_$(MACHDEP)_$(MULTIARCH).py \
 		$(DESTDIR)$(LIBDEST); \
 	$(INSTALL_DATA) $(srcdir)/LICENSE $(DESTDIR)$(LIBDEST)/LICENSE.txt
+	@ # If app store compliance has been configured, apply the patch to the
+	@ # installed library code. The patch has been previously validated against
+	@ # the original source tree, so we can ignore any errors that are raised
+	@ # due to files that are missing because of --disable-test-modules etc.
+	@if [ "$(APP_STORE_COMPLIANCE_PATCH)" != "" ]; then \
+		echo "Applying app store compliance patch"; \
+		patch --force --reject-file "$(abs_builddir)/app-store-compliance.rej" --strip 2 --directory "$(DESTDIR)$(LIBDEST)" --input "$(abs_srcdir)/$(APP_STORE_COMPLIANCE_PATCH)" || true ; \
```

Comment:
Some explanation of the choices here:
* `--force` means no questions are asked of the user, and the patch is assumed to be forward.
* `--reject-file` provides a file where rejected chunks will be output. 

The `--reject-file` option is required because installing when `--disable-test-modules` is enabled results in a failure patching `test/test_urlparse.py` - which then fails again because the `test` folder doesn't exist, so it can't write a `.rej` file. If the directory *did* exist, there would be a `.rej` file in the distributed artefact. 

`.orig` files shouldn't be generated because we know the patch applies cleanly as a result of pre-validation.